File: abelkhan_type_csharp_component_msgpack/component/dbproxy/center_msg_handle.cs
/*
 * center_msg_handle
 * 2020/6/3
 * qianqians
 */

namespace abelkhan
{
	public class center_msg_handle
	{
		private center_call_server_module _center_call_server_module;
		private closeHandle _closehandle;
		private centerproxy _centerproxy;
		private hubmanager _hubs;

		public center_msg_handle(modulemng modules, hubmanager _hubmanager, closeHandle _closeHandle, centerproxy _proxy)
		{
			_hubs = _hubmanager;
			_closehandle = _closeHandle;
			_centerproxy = _proxy;

			_center_call_server_module = new center_call_server_module(modules);
			_center_call_server_module.onclose_server += close_server;
			_center_call_server_module.onreg_server_sucess += reg_server_sucess;
			_center_call_server_module.onserver_be_close += on_svr_closed;
		}

		public void close_server()
		{
			log.trace(new System.Diagnostics.StackFrame(true), timerservice.Tick, "close_server");
			_closehandle.is_close = true;
		}

		public void reg_server_sucess()
		{
			_centerproxy.is_reg_sucess = true;
		}

		public void on_svr_closed(string type, string name)
        {
			log.trace(new System.Diagnostics.StackFrame(true), timerservice.Tick, "on_svr_closed type:{0}, name:{1}", type, name);
			_hubs.on_hub_closed(name);
		}
	}
}

File: abelkhan_type_component/component/dbproxy/center_msg_handle.ts
import abelkhan = require('../../abelkhan_type/ts/abelkhan');
import center = require('../../protcol/ts/center');
import closehandle = require('./closehandle');
import centerproxy = require('./centerproxy');

export class center_msg_handle{
	private center_call_server_module : center.center_call_server_module;
	private _close_handle : closehandle.closeHandle;
	private _centerproxy : centerproxy.centerproxy;
	constructor(modules:abelkhan.modulemng, _close_handle:closehandle.closeHandle, _centerproxy:centerproxy.centerproxy){
		this._close_handle = _close_handle;
		this._centerproxy = _centerproxy;

		this.center_call_server_module = new center.center_call_server_module(modules);
		this.center_call_server_module.cb_close_server = this.close_server.bind(this);
		this.center_call_server_module.cb_reg_server_sucess = this.reg_server_sucess.bind(this);
	}

	public close_server(){
		this._close_handle._is_close = true;
	}

	public reg_server_sucess(){
		this._centerproxy.is_reg_sucess = true;
	}
}
